Sources/main.cpp

```diff
@@ -345,50 +377,42 @@ static int GdbserverMain(int argc, char **argv) {
     opts.usageDie("either a program or target PID is required in gdb mode");
   }
 
-  // This is used for llgs testing. We determine a port number dynamically and
-  // write it back to the FIFO passed as argument for the test harness to use
-  // it.
-  std::string namedPipePath = opts.getString("named-pipe");
-
   bool reverse = opts.getBool("reverse-connect");
 
   std::unique_ptr<Socket> socket;
 
+  switch (connection_type) {
+  case channel_type::named_pipe:
 #if defined(OS_POSIX)
-  if (fd >= 0) {
-    socket = CreateFDSocket(fd);
-  } else {
+    if (fd >= 0)
+      DS2LOG(Error, "named pipe should not be used with fd option");
 #endif
-    if (opts.getPositional("[host]:port").empty()) {
-      // If we have a named pipe, set the port to 0 to indicate that we should
-      // dynamically allocate it and write it back to the FIFO.
-      socket = CreateTCPSocket(
-          gDefaultHost, namedPipePath.empty() ? gDefaultPort : "0", reverse);
-    } else {
-      socket = CreateSocket(opts.getPositional("[host]:port"), reverse);
+    [[fallthrough]];
+  case channel_type::network:
+    // If we have a named pipe, set the port to 0 to indicate that we should
```

Comment:
since this code now also handles network, the comment is slightly outdated as it only mentions "named pipe".